### PR TITLE
IT-35042 / Externalize Service List into Json File in Gradlehome

### PR DESCRIPTION
**Pull request by apgsga-jhe**
List of supported services is retrieved from an external JSON file called supportedServices.json.

It has been made based on the same logic as for repoNames.json.

### CODE REVIEW

File: common-service-packager/src/main/java/com/apgsga/gradle/common/pkg/extension/ApgCommonPackageExtension.java [hunk 1]
```diff
@@ -61,6 +56,9 @@ public ApgCommonPackageExtension(Project project) {
 		super();
 		this.project = project;
 	}
+
```
Comment (chhex) [inline common-service-packager/src/main/java/com/apgsga/gradle/common/pkg/extension/ApgCommonPackageExtension.java]: @apgsga-jhe I would get rid of those on the public api. 
They should'nt be exposed on the DSL.  If you need those you make them protected

File: common-service-packager/src/main/java/com/apgsga/gradle/common/pkg/plugin/ApgCommonPackagePlugin.java [hunk 2]
```diff
@@ -8,27 +13,29 @@
...
 	public void apply(Project project) {
-
+		this.project = project;
 		final ExtensionContainer ext = project.getExtensions();
 		final Logger logger = project.getLogger();
 		final PluginContainer plugins = project.getPlugins();
 		plugins.apply(ApgRepoConfigPlugin.class);
-		ext.create("apgPackage", ApgCommonPackageExtension.class, project);
+		ApgCommonPackageExtension apgPackage = ext.create("apgPackage", ApgCommonPackageExtension.class, project);
+		apgPackage.setSupportedServices(loadSupportedServices().supportedServices);
```
Comment (chhex) [inline common-service-packager/src/main/java/com/apgsga/gradle/common/pkg/plugin/ApgCommonPackagePlugin.java]: @apgsga-jhe  Ok, here you need the setter , that's fine.  i would make even concessions to the packaging to have them protected

**Comment by apgsga-jhe** [comment]
@chhex : I re-requested a review ;) .... agreed with the above comment, the setter is now protected ....

**Comment by apgsga-jhe** [comment]
> Supi

@chhex  : Actually I went a bit fast :) .... naturally, the plugin then can't access anymore the setter which is in a different package ... I'll change it again and pass the list as a "ApgCommonPackageExtension constructor's parameter" ... except if I missed something ?!?

**Comment by apgsga-jhe** [comment]
@chhex  .... Ok, here my last change, I'll merge: https://github.com/apgsga-it/apg-gradle-plugins/pull/17/commits/3be4ce5d1e98e96bb76b6083e6ba816c93f7db40

**Pull request merged**